lib/python/qmk/keymap.py

```diff
@@ -19,6 +19,19 @@
 };
 """
 
+ENCODER_SUPPORT = """void encoder_update_user(uint8_t index, bool clockwise) {
+__ENCODERS_GO_HERE__
+};
+"""
+
+ENCODER_IF = """\t_ELIF_ (index == _INDEX_) {
+\t\tif (clockwise) {
+\t\t\ttap_code16(_CLOCKWISE_);
+\t\t} else {
+\t\t\ttap_code16(_COUNTER_);
+\t\t}
+\t}"""
+
```

Comment:
As per my previous comment, this will need some logic with some checks to see what layer we're currently in and using the appropriate code.